src/modules/facility/dto/create-facility-request.dto.ts

```diff
@@ -0,0 +1,190 @@
+import { EXAMPLES, PROPERTIES } from '@ukef/constants';
+import { ValidatedCurrencyApiProperty } from '@ukef/decorators/validated-currency-api-property-decorator';
+import { ValidatedDateOnlyApiProperty } from '@ukef/decorators/validated-date-only-api-property.decorator';
+import { ValidatedDealIdentifierApiProperty } from '@ukef/decorators/validated-deal-identifier-api-property.decorator';
+import { ValidatedFacilityIdentifierApiProperty } from '@ukef/decorators/validated-facility-identifier-api-property';
+import { ValidatedNumberApiProperty } from '@ukef/decorators/validated-number-api-property.decorator';
+import { ValidatedPartyIdentifierApiProperty } from '@ukef/decorators/validated-party-identifier-api-property.decorator';
+import { ValidatedStringApiProperty } from '@ukef/decorators/validated-string-api-property.decorator';
+import { DateOnlyString } from '@ukef/helpers';
+
+export type CreateFacilityRequest = CreateFacilityRequestItem[];
+
+// TODO APIM-69: description improvements if possible
+export class CreateFacilityRequestItem {
+  @ValidatedDealIdentifierApiProperty({
+    description: 'The identifier of the deal for the facility.',
+  })
+  dealIdentifier: string;
+
+  @ValidatedFacilityIdentifierApiProperty({
+    description: 'The identifier of the facility to create.',
+  })
+  facilityIdentifier: string;
+
+  @ValidatedStringApiProperty({
+    description: 'The identifier of the portfolio that owns the facility.',
+    example: PROPERTIES.GLOBAL.portfolioIdentifier,
+    required: false,
+    default: PROPERTIES.GLOBAL.portfolioIdentifier,
+    maxLength: 2,
+  })
+  portfolioIdentifier?: string; // TODO APIM-69: is it right to allow the user to specify a different portfolio id?
```

Comment:
We have just one portfolio and don't think that this will change. So all records needs to be in this E1 portfolio.

Mulesoft is expecting portfolioIdentifier for most POST calls (optional in most cases). At the moment it is for client to ensure that this ID doesn't change then creating Deal, Facility or their child records.

Because we always had just 1 Portfolio we never fully tested if having another would work correctly.

Reply:
Okay thanks Audrius, I'll remove this field